IM/connectors/OCCI.py

```diff
@@ -1333,5 +1388,51 @@ def get_keystone_token(occi, keystone_uri, auth):
 
             return tenant_token_id
         except Exception as ex:
-            occi.logger.exception("Error obtaining Keystone Token.")
-            raise Exception("Error obtaining Keystone Token: %s" % str(ex))
+            occi.logger.exception("Error obtaining Keystone v2 Token.")
+            raise Exception("Error obtaining Keystone v2 Token: %s" % str(ex))
+
+    @staticmethod
+    def get_keystone_token_v3(occi, keystone_uri, auth):
+        """
+        Contact the specified keystone v3 server to return the token
+        """
+        try:
+            uri = uriparse(keystone_uri)
+            server = uri[1].split(":")[0]
+            port = int(uri[1].split(":")[1])
+
+            headers = {'Accept': 'application/json', 'Connection': 'close', 'Content-Type': 'application/json'}
+
+            if auth and "token" in auth:
+                # Use OpenID
+                url = "https://%s:%s/v3/OS-FEDERATION/identity_providers/egi.eu/protocols/oidc/auth" % (server, port)
+            else:
+                # Use VOMS proxy
+                url = "https://%s:%s/v3/OS-FEDERATION/identity_providers/egi.eu/protocols/mapped/auth" % (server, port)
+
+            resp = occi.create_request_static('POST', url, auth, headers)
```

Comment:
Could you, please, use `GET` instead of `POST` when requesting an unscoped token? Native OS implementations will accept both, but the official API doc says:
```
GET | /v3/OS-FEDERATION/identity_providers/{idp_id}/protocols/{protocol_id}/auth | Obtains an unscope token using federation authentication.
```
and our implementation is more conservative and will refuse POST requests.